benchmark/benchmark.R

```diff
@@ -0,0 +1,179 @@
+  cat("Expected Log Change Yeast:", yeast_median, "\n")
+  
+  #calculate SD and mean
+  
+  
+  # Kept the code for Individual Boxplots
+  
+  # boxplot(human_comparisonResult$log2FC,
+  #         main = "Boxplot of log2FC for Human",
+  #         ylab = "log2FC",
+  #         col = "lightblue")
+  # 
+  # 
+  boxplot(ecoli_comparisonResult$log2FC,
+          main = "Boxplot of log2FC for E. coli",
+          ylab = "log2FC",
+          col = "lightgreen")
+  # 
+  # boxplot(yeast_comparisonResult$log2FC,
+  #         main = "Boxplot of log2FC for Yeast",
+  #         ylab = "log2FC",
+  #         col = "lightpink")
+  
+  combined_data <- list(
+    Human = human_comparisonResult$log2FC,
+    Ecoli = ecoli_comparisonResult$log2FC,
+    Yeast = yeast_comparisonResult$log2FC
+  )
+  
+  
+  unique_ecoli_proteins <- unique(ecoli_comparisonResult$Protein)
+  unique_yeast_proteins <- unique(yeast_comparisonResult$Protein)
+  
+  all_proteins <- c(union(unique_ecoli_proteins, unique_yeast_proteins))  # find out the significant proteins in FragData
+  
+  extracted_proteins <- sapply(all_proteins, function(x) {
+    split_string <- strsplit(x, "\\|")[[1]]  # Split the string by '|'
+    if (length(split_string) >= 2) {
+      return(split_string[2])  # Return the second element
+    } else {
+      return(NA)  # Return NA if there's no second element
+    }
+  })
+  
+  extracted_proteins <- unname(unlist(extracted_proteins))
+  
+  proteins <- c(extracted_proteins)
+  
+  
+  TP <- comparisonResult %>% filter(grepl(paste(proteins, collapse = "|"), Protein) & adj.pvalue < 0.05) %>% nrow()
+  
+  
+  FP <- comparisonResult %>% filter(!grepl(paste(proteins, collapse = "|"), Protein) & adj.pvalue < 0.05) %>% nrow()
+  
+  
+  TN <- comparisonResult %>% filter(!grepl(paste(proteins, collapse = "|"), Protein) & adj.pvalue >= 0.05) %>% nrow()
+  
+  
+  FN <- comparisonResult %>% filter(grepl(paste(proteins, collapse = "|"), Protein) & adj.pvalue >= 0.05) %>% nrow()
+  
+  cat("True Positives (Yeast and EColi): ", TP, "\n")
+  cat("False Positives (Human Samples)", FP, "\n")
+  cat("True Negatives", TN, "\n")
+  cat("False Negatives", FN, "\n")
+  
+  FPR <- FP / (FP + TN)
+  
+  # Accuracy
+  accuracy <- (TP + TN) / (TP + TN + FP + FN)
+  
+  # Recall
+  recall <- TP / (TP + FN)
+  
+  results <- data.frame(
+    Label = label,
+    TP = TP,
+    FP = FP,
+    TN = TN,
+    FN = FN,
+    FPR = FPR,
+    Accuracy = accuracy,
+    Recall = recall
+  )
+  
+  return(results)
+  
+}
+
+start_time <- Sys.time()
+
+# Use fread directly to read the CSV
+fragpipe_raw = data.table::fread("..//data//FragPipeMsStatsBenchmarking.csv")
+
+head(fragpipe_raw)
+
+fragpipe_raw$Condition = unlist(lapply(fragpipe_raw$Run, function(x){
+  paste(str_split(x, "\\_")[[1]][4:5], collapse="_")
+}))
+
+fragpipe_raw$BioReplicate = unlist(lapply(fragpipe_raw$Run, function(x){
+  paste(str_split(x, "\\_")[[1]][4:7], collapse="_")
+}))
```

Comment:
I thought the Fragpipe files already have BioReplicate and Condition information

Reply:
Could you use the dataset stored in the `/work/VitekLab/Data/MS/Benchmarking` directory?  It should have the name `MSstats.csv`